server/dex/dex.go

```diff
@@ -882,9 +882,75 @@ func (dm *DEX) EpochOrders(base, quote uint32) ([]order.Order, error) {
 	return dm.storage.EpochOrders(base, quote)
 }
 
+// MatchData embeds db.MatchData with decoded swap transaction coin IDs.
+type MatchData struct {
+	db.MatchData
+	MakerSwap   string
+	TakerSwap   string
+	MakerRedeem string
+	TakerRedeem string
+}
+
 // MarketMatches returns matches for market with base and quote.
-func (dm *DEX) MarketMatches(base, quote uint32, includeInactive bool) ([]*db.MatchData, error) {
-	return dm.storage.MarketMatches(base, quote, includeInactive)
+func (dm *DEX) MarketMatches(base, quote uint32, includeInactive bool) ([]*MatchData, error) {
+	baseAsset := dm.assets[base]
+	if baseAsset == nil {
+		return nil, fmt.Errorf("asset %d not found", base)
+	}
+	quoteAsset := dm.assets[quote]
+	if quoteAsset == nil {
+		return nil, fmt.Errorf("asset %d not found", quote)
+	}
+	mds, err := dm.storage.MarketMatches(base, quote, includeInactive)
+	if err != nil {
+		return nil, err
+	}
+
+	matchDatas := make([]*MatchData, 0, len(mds))
+	for _, md := range mds {
+		matchData := MatchData{
+			MatchData: md.MatchData,
+		}
+		// asset0 is the maker swap / taker redeem asset.
+		// asset1 is the taker swap / maker redeem asset.
+		// Maker selling means asset 0 is base; asset 1 is quote.
+		asset0, asset1 := baseAsset, quoteAsset
+		if md.TakerSell {
+			asset0, asset1 = quoteAsset, baseAsset
+		}
+		if len(md.MakerSwapCoin) > 0 {
+			coinStr, err := asset0.Backend.ValidateCoinID(md.MakerSwapCoin)
+			if err != nil {
+				log.Errorf("Unable to decode coin %x: %v", md.MakerSwapCoin, err)
```

Comment:
I can only see this happening if coin ID encodings change or this is for a different asset, but I don't want to it refuse to response with the data it can decode.